tools/src/main/java/org/semanticweb/owlapi/modularity/AbstractModuleExtractor.java

```diff
@@ -22,203 +17,77 @@
  *
  */
 public abstract class AbstractModuleExtractor implements ModuleExtractor {
-
+	
 	/**
 	 * The axiom base of this {@link AbstractModuleExtractor}.
 	 */
 	private final @Nonnull Set<OWLAxiom> axiomBase;
-
+	
 	/**
 	 * The axioms that are guaranteed to be contained in every module calculated by
 	 * this {@link AbstractModuleExtractor}. Will be calculated when first calling
-	 * {@link AbstractModuleExtractor#precomputeGlobals()}.
+	 * {@link AbstractModuleExtractor#globals()}.
 	 */
-	private @Nonnull Optional<Set<OWLAxiom>> globals;
-
+	private @Nonnull Optional<Set<OWLAxiom>> globals = Optional.empty();
+	
 	/**
 	 * The axioms that are guaranteed to be contained in no module calculated by
 	 * this {@link AbstractModuleExtractor}. Will be calculated when first calling
-	 * {@link AbstractModuleExtractor#precomputeTautologies()}.
+	 * {@link AbstractModuleExtractor#tautologies()}.
 	 */
-	private @Nonnull Optional<Set<OWLAxiom>> tautologies;
-
+	private @Nonnull Optional<Set<OWLAxiom>> tautologies = Optional.empty();
+	
 	/**
 	 * Creates a new {@link AbstractModuleExtractor}.
 	 *
 	 * @param axiomBase the axiom base of the new {@link AbstractModuleExtractor}
 	 */
 	protected AbstractModuleExtractor(final Stream<OWLAxiom> axiomBase) {
 		this.axiomBase = axiomBase.filter(OWLAxiom::isLogicalAxiom).collect(Collectors.toSet());
-		globals = Optional.empty();
-		tautologies = Optional.empty();
 	}
-
-	/**
-	 * Return the axioms all modules of this {@link ModuleExtractor} are computed
-	 * against, including global axioms and tautologies. Subclasses should use
-	 * {@link AbstractModuleExtractor#precomputedAxioms()} which does not contain
-	 * global axioms and tautologies (if they are precomputed, respectively).
-	 *
-	 * @return The axioms as specified above
-	 */
+	
 	@Override
 	public final @Nonnull Stream<OWLAxiom> axiomBase() {
 		Stream<OWLAxiom> toReturn = axiomBase.stream();
 		toReturn = Stream.concat(tautologies.stream().flatMap(Set::stream), toReturn);
-		return Stream.concat(globals.stream().flatMap(Set::stream), toReturn);
+		return axiomBase.stream();
```

Comment:
Was this the intention? globals is no longer involved and the stream creation with tautologies and base axioms is unused